src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestRespondentEvidencePreparer.java

```diff
@@ -121,12 +127,11 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
         int dueInDays = HandlerUtils.isAcceleratedDetainedAppeal(asylumCase)
             ? requestRespondentEvidenceDueInDaysAda
             : requestRespondentEvidenceDueInDays;
-        asylumCase.write(SEND_DIRECTION_DATE_DUE,
-            dateProvider
-                .now()
-                .plusDays(dueInDays)
-                .toString()
-        );
+        LocalDate dueDate = HandlerUtils.isAcceleratedDetainedAppeal(asylumCase)
+                ? dueDateService.calculateDueDate(dateProvider.now().atStartOfDay(ZoneOffset.UTC), dueInDays).toLocalDate()
+                : dateProvider.now().plusDays(dueInDays);
```

Comment:
You can actually use the "requestRespondentEvidenceDueInDaysAda" and "requestRespondentEvidenceDueInDays" directly in calculating the dueDate since you are checking for ADA again. So you can get rid of the variable "dueInDays"